index.js

```diff
@@ -544,21 +544,48 @@ alexa.app = function(name, endpoint) {
     return self.handler;
   };
 
-  // A utility method to bootstrap alexa endpoints into express automatically
-  this.express = function(express, path, enableDebug) {
-    var endpoint = (path || "/") + (self.endpoint || self.name);
-    express.post(endpoint, function(req, res) {
+  // attach Alexa endpoint to an express router
+  //
+  // @param object options.expressApp the express instance to attach to
+  // @param router options.express router instance to attach to the express app
+  // @param string options.endpoint the path to attach the router to (e.g., passing 'mine' attaches to '/mine')
+  // @param bool options.checkCert when true, applies Alexa certificate checking (default true)
+  // @param bool options.debug when true, sets up the route to handle GET requests (default false)
+  // @throws Error when router or expressApp options are not specified
+  // @return ? TODO: not sure what this returns
+  this.express = function(options) {
+    if (!options.expressApp) {
+      throw new Error("You must specify an express instance to attach to.");
+    }
+
+    if (!options.router) {
+      throw new Error("You must specify an express router to attach.");
+    }
+
+    var defaultOptions = { endpoint: self.name, checkCert: true, debug: false };
+
+    options = defaults(options, defaultOptions);
+
+    var endpoint = "/" + options.endpoint;
+    var router = options.router;
+
+    options.expressApp.use(endpoint, router);
+
+    if (options.checkCert) {
+      options.router.use(verifierMiddleware({ strictHeaderCheck: true }));
+    }
+
+    // exposes POST /<endpoint> route
+    router.post("/", function(req, res) {
       self.request(req.body).then(function(response) {
         res.json(response);
       }, function() {
         res.status(500).send("Server Error");
       });
     });
-    if (typeof enableDebug != "boolean") {
-      enableDebug = true;
-    }
-    if (enableDebug) {
-      express.get(endpoint, function(req, res) {
+
+    if (options.debug) {
+      router.get("/", function(req, res) {
```

Comment:
One issue that I just found while testing. When you do a GET request and the verifier middleware is enabled with strict header checking, it gives me a 401 error, which shouldn't happen at all.

One possible fix is that the alexa-verifier-middleware should check what request method is being sent and if it is a POST request, then process the request accordingly. [Here](https://expressjs.com/en/4x/api.html#req.method) is a way to access the request's method from expressjs.